components/deployment-automation/org.wso2.carbon.deployment.automation/src/main/java/org/wso2/carbon/deployment/automation/DeploymentService.java

```diff
@@ -0,0 +1,198 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.deployment.automation;
+
+import io.swagger.annotations.*;
+import org.osgi.service.component.annotations.Component;
+import org.wso2.carbon.deployment.automation.exceptions.DeploymentAutomationException;
+import org.wso2.carbon.deployment.automation.interfaces.DeploymentProvider;
+import org.wso2.carbon.deployment.automation.kubernetes.KubernetesDeploymentProvider;
+import org.wso2.carbon.deployment.automation.models.Deployment;
+import org.wso2.msf4j.Microservice;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+
+/**
+ * Handles deployment automation in a containerized environment. The service is available at,
+ * http://localhost:9090/deployments
+ */
+@Api(value = "service", description = "Manage deployment automation in a containerized environment")
+@SwaggerDefinition(
+        info = @Info(
+                title = "Deployment Automation Swagger Definition",
+                version = "1.0",
+                description = "Deployment automation service. Manages deployments in a containerized environment.",
+                license = @License(
+                        name = "Apache 2.0",
+                        url = "http://www.apache.org/licenses/LICENSE-2.0"
+                ),
+                contact = @Contact(
+                        name = "WSO2 Inc.",
+                        email = "dev@wso2.org",
+                        url = "http://wso2.com"
+                )
+        )
+)
+@Path("/deployments")
+@Component(
+        name = "org.wso2.carbon.deployment.automation.DeploymentService",
+        service = Microservice.class,
+        immediate = true
+)
+public class DeploymentService implements Microservice {
+    private static final String DEPLOYMENT_KUBERNETES = "kubernetes";
+
+    /**
+     * Get list of deployments.
+     * http://localhost:9090/deployments?platform=kubernetes
+     *
+     * @param platform Deployment platform
+     * @return Response
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/")
+    @ApiOperation(
+            value = "Get all deployments",
+            notes = "Returns all deployment",
+            response = Deployment.class,
+            responseContainer = "List")
+    public Response getDeployments(@QueryParam("platform") String platform) {
+        try {
+            List<Deployment> deployments = getDeploymentProvider(platform).listDeployments();
+            return Response.ok()
+                    .entity(deployments)
+                    .build();
+        } catch (DeploymentAutomationException e) {
+            return Response.status(Response.Status.BAD_REQUEST)
```

Comment:
According to this logic every time a DeploymentAutomationException is raised client will receive HTTP 400. As a result, developers would need to know this mapping been the exception and the response status code. I think to make it more readable we can rename the exception to BadRequestException.

Moreover, I think it might be better to use an Exception Mapper for this scenario:
https://docs.wso2.com/display/MSF4J200/Sending+Customized+Responses+for+Exceptions